src/android/NotificationHandler.java

```diff
@@ -0,0 +1,115 @@
+package com.spoon.backgroundfileupload;
+
+import android.app.Activity;
+import android.app.NotificationManager;
+import android.content.res.Resources;
+import android.widget.RemoteViews;
+
+import androidx.core.app.NotificationCompat;
+
+import net.gotev.uploadservice.UploadService;
+import net.gotev.uploadservice.observer.task.AbstractSingleNotificationHandler;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Map;
+
+public class NotificationHandler extends AbstractSingleNotificationHandler {
+
+    private Activity mContext;
+
+    public NotificationHandler(@NotNull UploadService service, Activity context) {
+        super(service);
+        this.mContext = context;
+    }
+
+    @Nullable
+    @Override
+    public NotificationCompat.Builder updateNotification(@NotNull NotificationManager notificationManager, @NotNull NotificationCompat.Builder builder, @NotNull Map<String, TaskData> map) {
+        int speed = 0;
+        int inProgress = 0;
+
+        for (Map.Entry<String, TaskData> entry : map.entrySet()) {
+            String uploadId = entry.getValue().getInfo().getUploadId();
+
+            if (entry.getValue().getStatus() == TaskStatus.InProgress) {
+                inProgress++;
+
+                speed += convertUnit(
+                        entry.getValue().getInfo().getUploadRate().getUnit().name(),
+                        entry.getValue().getInfo().getUploadRate().getValue()
+                );
+            }
+
+            if (entry.getValue().getStatus() == TaskStatus.Failed || entry.getValue().getStatus() == TaskStatus.Succeeded) {
+                removeTask(uploadId);
+            }
```

Comment:
```suggestion
            if (entry.getValue().getStatus() == TaskStatus.Failed || entry.getValue().getStatus() == TaskStatus.Succeeded) {
                removeTask(uploadId);
            }
```

Ce qui serait plus propre, c'est de ne pas faire ça ici, mais le faire dans le onComplete non ?

```
    override fun onCompleted(
        info: UploadInfo,
        notificationId: Int,
        notificationConfig: UploadNotificationConfig
    ) {
    }
```

Reply:
oui j'y ai pensé aussi mais faudra que je fasse une PR à l'auteur de la lib